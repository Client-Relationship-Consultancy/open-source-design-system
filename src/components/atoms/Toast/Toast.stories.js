import React from "react"
import { storiesOf } from "@storybook/react"
import { boolean } from "@storybook/addon-knobs"
import { Toast } from "../../../index"
import { styled } from "@storybook/theming"

const Container = styled.div`
  position: relative;
  width: 100%;
  height: 500px;
`

const stories = storiesOf("Toast", module)

stories.add("Standard style", (context) => {
  const props = ""

  Object.assign(context, props)

  return (
    <Container>
      <Toast visible={boolean("show toast", true)}>
        <div style={{ display: "flex", flexDirection: "row", whiteSpace: "nowrap" }}>
          <svg
            width="24"
            height="17"
            viewBox="0 0 24 17"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.88403 15.97V14.97H8.83662L8.78942 14.9745L8.88403 15.97ZM7.87395 5.79577L7.72107 6.78401L8.65054 6.92781L8.85098 6.00889L7.87395 5.79577ZM14.2547 1.08321L14.4115 0.0955903L14.2547 1.08321ZM19.0587 6.76857L18.0616 6.69281L17.9841 7.71238L19.0052 7.76713L19.0587 6.76857ZM11.7076 6.79289L11.0005 6.08579L9.58628 7.5L10.2934 8.20711L11.7076 6.79289ZM14.7934 12.7071L15.5005 13.4142L16.9147 12L16.2076 11.2929L14.7934 12.7071ZM16.2076 8.20711L16.9147 7.5L15.5005 6.08579L14.7934 6.79289L16.2076 8.20711ZM10.2934 11.2929L9.58628 12L11.0005 13.4142L11.7076 12.7071L10.2934 11.2929ZM8.88403 15.97C8.78942 14.9745 8.78976 14.9745 8.79007 14.9745C8.79013 14.9745 8.79041 14.9744 8.79053 14.9744C8.79079 14.9744 8.79092 14.9744 8.79094 14.9744C8.79098 14.9744 8.79055 14.9744 8.78967 14.9745C8.7879 14.9746 8.7843 14.975 8.77892 14.9754C8.76818 14.9763 8.75037 14.9777 8.72597 14.9794C8.67714 14.9828 8.60209 14.9874 8.50459 14.9913C8.30926 14.9993 8.02573 15.0048 7.68388 14.9939C6.99404 14.9721 6.09654 14.8847 5.21642 14.6338C4.3314 14.3815 3.52888 13.9824 2.95489 13.3824C2.40364 12.8062 2.00049 11.9811 2.00049 10.7258H0.000488281C0.000488281 12.4672 0.582779 13.796 1.50969 14.765C2.41387 15.7101 3.58223 16.2476 4.66809 16.5571C5.75886 16.8681 6.83225 16.968 7.62056 16.9929C8.0178 17.0055 8.35017 16.9993 8.58598 16.9897C8.70404 16.9849 8.79836 16.9792 8.86501 16.9746C8.89835 16.9722 8.92481 16.9702 8.9439 16.9686C8.95344 16.9678 8.96115 16.9671 8.96696 16.9666C8.96986 16.9664 8.97229 16.9661 8.97424 16.966C8.97522 16.9659 8.97607 16.9658 8.9768 16.9657C8.97717 16.9657 8.97763 16.9656 8.97781 16.9656C8.97823 16.9656 8.97863 16.9655 8.88403 15.97ZM2.00049 10.7258C2.00049 9.43166 2.36564 8.61766 2.82999 8.08216C3.30776 7.53117 3.97095 7.18323 4.70799 6.97591C5.44547 6.76848 6.2028 6.71771 6.7906 6.72322C7.081 6.72595 7.3216 6.74229 7.48658 6.75754C7.56888 6.76515 7.63181 6.77243 7.6721 6.77749C7.69223 6.78002 7.70666 6.78199 7.71497 6.78316C7.71912 6.78374 7.72174 6.78412 7.72278 6.78428C7.72329 6.78435 7.72341 6.78437 7.72313 6.78433C7.72299 6.78431 7.72275 6.78427 7.72241 6.78422C7.72223 6.78419 7.7219 6.78414 7.72181 6.78413C7.72145 6.78407 7.72107 6.78401 7.87395 5.79577C8.02684 4.80753 8.0264 4.80746 8.02594 4.80739C8.02575 4.80736 8.02525 4.80728 8.02487 4.80722C8.02411 4.80711 8.02324 4.80697 8.02227 4.80683C8.02032 4.80653 8.01795 4.80618 8.01517 4.80577C8.0096 4.80494 8.00236 4.80389 7.99351 4.80265C7.97581 4.80016 7.95163 4.79689 7.92142 4.79309C7.86102 4.7855 7.77628 4.77579 7.67064 4.76603C7.45973 4.74654 7.16334 4.72663 6.80935 4.72331C6.10837 4.71674 5.14734 4.77472 4.16645 5.05062C3.18513 5.32665 2.12995 5.83663 1.31895 6.7719C0.494521 7.72267 0.000488281 9.02327 0.000488281 10.7258H2.00049ZM7.87395 5.79577C8.85098 6.00889 8.85089 6.00931 8.8508 6.0097C8.85078 6.00981 8.8507 6.01019 8.85065 6.0104C8.85056 6.01082 8.85048 6.01117 8.85042 6.01143C8.85031 6.01195 8.85026 6.01215 8.85029 6.01203C8.85034 6.01179 8.85068 6.01028 8.85132 6.00755C8.8526 6.00208 8.85507 5.99172 8.8588 5.97684C8.86626 5.94704 8.87875 5.89925 8.89688 5.83629C8.93323 5.71007 8.99174 5.52459 9.07713 5.30189C9.24951 4.8523 9.52293 4.27332 9.92788 3.72647C10.7122 2.66736 11.9662 1.73229 14.0978 2.07083L14.4115 0.0955903C11.3696 -0.387541 9.43328 1.03366 8.32059 2.53625C7.77705 3.27025 7.42533 4.02347 7.20969 4.58588C7.10107 4.86917 7.02486 5.10964 6.97498 5.28287C6.94999 5.36962 6.9315 5.43993 6.91875 5.4908C6.91238 5.51625 6.90743 5.53687 6.90383 5.55229C6.90202 5.56 6.90055 5.56642 6.89941 5.5715C6.89884 5.57403 6.89834 5.57624 6.89793 5.5781C6.89772 5.57903 6.89754 5.57987 6.89737 5.58063C6.89729 5.58101 6.89718 5.58152 6.89713 5.58171C6.89703 5.58219 6.89693 5.58265 7.87395 5.79577ZM14.0978 2.07083C16.3069 2.42167 17.2399 3.58655 17.6749 4.62154C17.9032 5.16464 18.0012 5.695 18.0405 6.09384C18.0599 6.29145 18.0645 6.45161 18.0645 6.55798C18.0645 6.61101 18.0634 6.65021 18.0624 6.67348C18.062 6.68511 18.0615 6.69271 18.0614 6.69603C18.0613 6.69769 18.0612 6.69827 18.0612 6.69775C18.0613 6.69748 18.0613 6.69694 18.0614 6.69612C18.0614 6.69571 18.0614 6.69523 18.0615 6.69468C18.0615 6.6944 18.0615 6.69393 18.0615 6.6938C18.0616 6.69331 18.0616 6.69281 19.0587 6.76857C20.0559 6.84433 20.0559 6.84378 20.0559 6.84323C20.056 6.84301 20.056 6.84243 20.056 6.84201C20.0561 6.84116 20.0562 6.84024 20.0562 6.83924C20.0564 6.83724 20.0565 6.83493 20.0567 6.83233C20.0571 6.82712 20.0575 6.82072 20.0579 6.81315C20.0588 6.79801 20.0598 6.7782 20.0608 6.754C20.0628 6.70563 20.0645 6.63953 20.0645 6.55797C20.0645 6.39515 20.0575 6.16892 20.0308 5.89772C19.9777 5.35894 19.8447 4.62202 19.5187 3.84649C18.8451 2.24423 17.3761 0.566425 14.4115 0.0955903L14.0978 2.07083ZM19.0587 6.76857C19.0052 7.76713 19.0046 7.7671 19.0041 7.76707C19.004 7.76707 19.0034 7.76704 19.0031 7.76702C19.0025 7.76698 19.0019 7.76695 19.0014 7.76692C19.0004 7.76686 18.9996 7.76681 18.999 7.76678C18.9979 7.76671 18.9976 7.76668 18.9982 7.76673C18.9994 7.76681 19.004 7.76715 19.0117 7.76787C19.0273 7.76929 19.0554 7.77218 19.0942 7.77743C19.1722 7.78796 19.2914 7.80772 19.4379 7.8434C19.7342 7.9156 20.121 8.04795 20.5003 8.28459C21.2049 8.72411 22.0005 9.5864 22.0005 11.475H24.0005C24.0005 8.86854 22.8252 7.37761 21.5589 6.58769C20.9528 6.2096 20.3542 6.00816 19.9114 5.90026C19.6883 5.84589 19.4996 5.81401 19.362 5.79543C19.293 5.78612 19.2364 5.78008 19.1943 5.77621C19.1732 5.77428 19.1556 5.77288 19.1419 5.77189C19.135 5.7714 19.1291 5.77101 19.1241 5.7707C19.1217 5.77054 19.1194 5.77041 19.1175 5.77029C19.1165 5.77023 19.1156 5.77018 19.1147 5.77013C19.1143 5.77011 19.1137 5.77008 19.1134 5.77006C19.1129 5.77003 19.1123 5.77 19.0587 6.76857ZM22.0005 11.475C22.0005 12.3664 21.7245 12.979 21.3446 13.4271C20.9491 13.8937 20.3942 14.2347 19.7768 14.4775C19.162 14.7193 18.533 14.8447 18.048 14.9083C17.8081 14.9397 17.6098 14.9552 17.4744 14.9628C17.4069 14.9666 17.3555 14.9684 17.3229 14.9693C17.3066 14.9697 17.2951 14.9699 17.2886 14.97C17.2854 14.97 17.2835 14.97 17.2828 14.97C17.2825 14.97 17.2825 14.97 17.2829 14.97C17.283 14.97 17.2833 14.97 17.2836 14.97C17.2838 14.97 17.2841 14.97 17.2842 14.97C17.2846 14.97 17.2849 14.97 17.2849 15.97C17.2849 16.97 17.2853 16.97 17.2858 16.97C17.2859 16.97 17.2864 16.97 17.2867 16.97C17.2874 16.97 17.2882 16.97 17.2891 16.97C17.2908 16.97 17.2929 16.97 17.2953 16.97C17.3002 16.97 17.3064 16.9699 17.314 16.9698C17.3292 16.9696 17.3497 16.9693 17.3753 16.9686C17.4265 16.9672 17.4978 16.9647 17.5866 16.9597C17.7638 16.9497 18.012 16.9301 18.3079 16.8913C18.8946 16.8144 19.6945 16.6589 20.5087 16.3388C21.3201 16.0197 22.1941 15.5179 22.8702 14.7203C23.5621 13.9042 24.0005 12.8311 24.0005 11.475H22.0005ZM8.88403 16.97H17.2849V14.97H8.88403V16.97ZM10.2934 8.20711L14.7934 12.7071L16.2076 11.2929L11.7076 6.79289L10.2934 8.20711ZM14.7934 6.79289L10.2934 11.2929L11.7076 12.7071L16.2076 8.20711L14.7934 6.79289Z"
              fill="#DC1010"
            />
          </svg>
          <p style={{ margin: 0, padding: 0, marginLeft: "7px" }}>
            Sorry, we werenâ€™t able to save your changes. Please try saving again
          </p>
          <p style={{ margin: 0, padding: 0, marginLeft: "7px", opacity: 0.6, cursor: "pointer" }}>
            try again
          </p>
        </div>
      </Toast>
    </Container>
  )
})
