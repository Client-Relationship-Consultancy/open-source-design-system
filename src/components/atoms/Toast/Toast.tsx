import React from "react";
import styled, { keyframes } from "styled-components";

enum Align {
  "center",
  "right",
  "left",
}
enum Position {
  "absolute",
  "fixed",
}
enum Type {
  "error",
  "success",
}

interface IToastContainerProps {
  align?: Align;
  top?: string;
  position?: Position;
  type?: Type;
  visible: boolean;
}

interface IToastProps extends IToastContainerProps {
  children: React.ReactNode;
}

const slideDown = keyframes`
  from {
    transform: translate(-50%,-100%);
    opacity:0;
  }
  to {
    transform: translate(-50%,0);
    opacity:1;
  }
`;
const slideUp = keyframes`
  from {
    opacity:1;
    transform: translate(-50%,0);
  }
  to {
    transform: translate(-50%,-100%);
    opacity:0;
  }
`;

export const ToastContainer = styled.div<IToastContainerProps>`
  align-content: "center";
  background-color: rgba(${(props) => props.theme.error.main.RGB}, 0.1);
  border: 0.0625rem solid rgba(${(props) => props.theme.error.main.RGB}, 0.5);
  box-sizing: border-box;
  border-radius: 0.25rem;
  position: ${(props: IToastContainerProps) => props.position || "absolute"};
  display: block;
  max-width: 100%;
  left: 50%;
  text-align: center;
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.375rem;
  top: ${(props: IToastContainerProps) => props.top || "0.625rem"};
  animation: ${(props: IToastContainerProps) => (props.visible ? slideDown : slideUp)} 0.3s forwards;
  padding: 0.5625rem 1rem;
  visibility: ${(props: IToastContainerProps) => (props.visible ? "visible" : "hidden")};
  transition: visibility 0.3s linear;
`;

ToastContainer.displayName = "ToastContainer";

const Toast: React.FunctionComponent<IToastProps> = (props) => {
  const { children } = props;
  return <ToastContainer {...props}>{children}</ToastContainer>;
};

Toast.displayName = "Toast";

export default Toast;
