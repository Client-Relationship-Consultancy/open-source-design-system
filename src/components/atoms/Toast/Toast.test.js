import React from "react"
import { mount, shallow } from "enzyme"
import toJson from "enzyme-to-json"
import "jest-styled-components"
import Toast, { ToastContainer } from "./Toast"

describe("Tabs", () => {
  const propsDefault = {
    align: "middle",
  top: undefined,
  position: undefined,
  type: "error",
  visible: true,
  }
  const props = {
    align: "middle",
    top: "20px",
    position: "fixed",
    type: "error",
    visible: true,
  }
  const propsNotVisible = {
    align: "middle",
    top: "20px",
    position: "fixed",
    type: "error",
    visible: false,
  }
  const componentDefault = shallow(<Toast {...propsDefault} ><div>test</div></Toast>)
  const component = shallow(
    <Toast {...props}>
      <div>test</div>
    </Toast>,
  )
  const componentnotVisible = shallow(
    <Toast {...propsNotVisible}>
      <div>test</div>
    </Toast>,
  )

  it("should match the snapshot for default props", () => {
    const tree = toJson(componentDefault)
    expect(tree).toMatchSnapshot()
  })

  it("should match the snapshot", () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it("should match the snapshot not visible", () => {
    const tree = toJson(componentnotVisible)
    expect(tree).toMatchSnapshot()
  })

  it("ToastContainer should match the snapshot default props", () => {
    const ToastContainerShallow = mount(
      <ToastContainer {...propsDefault}>
        <button className="selected" type="button">
          tab
        </button>
      </ToastContainer>,
    )
    const tree = toJson(ToastContainerShallow)
    expect(tree).toMatchSnapshot()
  })

  it("ToastContainer should match the snapshot", () => {
    const ToastContainerShallow = mount(
      <ToastContainer {...props}>
        <button className="selected" type="button">
          tab
        </button>
      </ToastContainer>,
    )
    const tree = toJson(ToastContainerShallow)
    expect(tree).toMatchSnapshot()
  })

  it("ToastContainer should match the snapshot not visible", () => {
    const ToastContainerShallow = mount(
      <ToastContainer {...propsNotVisible}>
        <button className="selected" type="button">
          tab
        </button>
      </ToastContainer>,
    )
    const tree = toJson(ToastContainerShallow)
    expect(tree).toMatchSnapshot()
  })

})
