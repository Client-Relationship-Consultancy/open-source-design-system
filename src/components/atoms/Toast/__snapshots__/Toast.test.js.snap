// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs ToastContainer should match the snapshot 1`] = `
.c0 {
  -webkit-align-content: "center";
  -ms-flex-line-pack: "center";
  align-content: "center";
  background-color: white;
  border: 0.0625rem solid rgba(220,16,16,0.5);
  box-sizing: border-box;
  border-radius: 0.25rem;
  position: fixed;
  display: block;
  max-width: 100%;
  left: 50%;
  text-align: center;
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.375rem;
  top: 20px;
  -webkit-animation: ezMiKc 0.3s forwards;
  animation: ezMiKc 0.3s forwards;
  visibility: visible;
  -webkit-transition: visibility 0.3s linear;
  transition: visibility 0.3s linear;
}

.c0 > div {
  width: 100%;
  height: 100%;
  background-color: rgba(220,16,16,0.1);
  padding: 0.5625rem 1rem;
}

<ToastContainer
  align="middle"
  position="fixed"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  top="20px"
  type="error"
  visible={true}
>
  <StyledComponent
    align="middle"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Toast__ToastContainer-y5ikly-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "align-content:\\"center\\";background-color:white;> div{width:100%;height:100%;background-color:rgba(",
            [Function],
            ",0.1);padding:0.5625rem 1rem;}border:0.0625rem solid rgba(",
            [Function],
            ",0.5);box-sizing:border-box;border-radius:0.25rem;position:",
            [Function],
            ";display:block;max-width:100%;left:50%;text-align:center;font-style:normal;font-weight:normal;font-size:0.875rem;line-height:1.375rem;top:",
            [Function],
            ";animation:",
            [Function],
            " 0.3s forwards;visibility:",
            [Function],
            ";transition:visibility 0.3s linear;",
          ],
        },
        "displayName": "ToastContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Toast__ToastContainer-y5ikly-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    position="fixed"
    theme={
      Object {
        "error": Object {
          "main": Object {
            "RGB": "220,16,16",
          },
        },
      }
    }
    top="20px"
    type="error"
    visible={true}
  >
    <div
      className="c0"
      type="error"
    >
      <button
        className="selected"
        type="button"
      >
        tab
      </button>
    </div>
  </StyledComponent>
</ToastContainer>
`;

exports[`Tabs ToastContainer should match the snapshot default props 1`] = `
.c0 {
  -webkit-align-content: "center";
  -ms-flex-line-pack: "center";
  align-content: "center";
  background-color: white;
  border: 0.0625rem solid rgba(220,16,16,0.5);
  box-sizing: border-box;
  border-radius: 0.25rem;
  position: absolute;
  display: block;
  max-width: 100%;
  left: 50%;
  text-align: center;
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.375rem;
  top: 0.625rem;
  -webkit-animation: ezMiKc 0.3s forwards;
  animation: ezMiKc 0.3s forwards;
  visibility: visible;
  -webkit-transition: visibility 0.3s linear;
  transition: visibility 0.3s linear;
}

.c0 > div {
  width: 100%;
  height: 100%;
  background-color: rgba(220,16,16,0.1);
  padding: 0.5625rem 1rem;
}

<ToastContainer
  align="middle"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  type="error"
  visible={true}
>
  <StyledComponent
    align="middle"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Toast__ToastContainer-y5ikly-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "align-content:\\"center\\";background-color:white;> div{width:100%;height:100%;background-color:rgba(",
            [Function],
            ",0.1);padding:0.5625rem 1rem;}border:0.0625rem solid rgba(",
            [Function],
            ",0.5);box-sizing:border-box;border-radius:0.25rem;position:",
            [Function],
            ";display:block;max-width:100%;left:50%;text-align:center;font-style:normal;font-weight:normal;font-size:0.875rem;line-height:1.375rem;top:",
            [Function],
            ";animation:",
            [Function],
            " 0.3s forwards;visibility:",
            [Function],
            ";transition:visibility 0.3s linear;",
          ],
        },
        "displayName": "ToastContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Toast__ToastContainer-y5ikly-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "error": Object {
          "main": Object {
            "RGB": "220,16,16",
          },
        },
      }
    }
    type="error"
    visible={true}
  >
    <div
      className="c0"
      type="error"
    >
      <button
        className="selected"
        type="button"
      >
        tab
      </button>
    </div>
  </StyledComponent>
</ToastContainer>
`;

exports[`Tabs ToastContainer should match the snapshot not visible 1`] = `
.c0 {
  -webkit-align-content: "center";
  -ms-flex-line-pack: "center";
  align-content: "center";
  background-color: white;
  border: 0.0625rem solid rgba(220,16,16,0.5);
  box-sizing: border-box;
  border-radius: 0.25rem;
  position: fixed;
  display: block;
  max-width: 100%;
  left: 50%;
  text-align: center;
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.375rem;
  top: 20px;
  -webkit-animation: goNLxH 0.3s forwards;
  animation: goNLxH 0.3s forwards;
  visibility: hidden;
  -webkit-transition: visibility 0.3s linear;
  transition: visibility 0.3s linear;
}

.c0 > div {
  width: 100%;
  height: 100%;
  background-color: rgba(220,16,16,0.1);
  padding: 0.5625rem 1rem;
}

<ToastContainer
  align="middle"
  position="fixed"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  top="20px"
  type="error"
  visible={false}
>
  <StyledComponent
    align="middle"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Toast__ToastContainer-y5ikly-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "align-content:\\"center\\";background-color:white;> div{width:100%;height:100%;background-color:rgba(",
            [Function],
            ",0.1);padding:0.5625rem 1rem;}border:0.0625rem solid rgba(",
            [Function],
            ",0.5);box-sizing:border-box;border-radius:0.25rem;position:",
            [Function],
            ";display:block;max-width:100%;left:50%;text-align:center;font-style:normal;font-weight:normal;font-size:0.875rem;line-height:1.375rem;top:",
            [Function],
            ";animation:",
            [Function],
            " 0.3s forwards;visibility:",
            [Function],
            ";transition:visibility 0.3s linear;",
          ],
        },
        "displayName": "ToastContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Toast__ToastContainer-y5ikly-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    position="fixed"
    theme={
      Object {
        "error": Object {
          "main": Object {
            "RGB": "220,16,16",
          },
        },
      }
    }
    top="20px"
    type="error"
    visible={false}
  >
    <div
      className="c0"
      type="error"
    >
      <button
        className="selected"
        type="button"
      >
        tab
      </button>
    </div>
  </StyledComponent>
</ToastContainer>
`;

exports[`Tabs should match the snapshot 1`] = `
<ToastContainer
  align="middle"
  position="fixed"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  top="20px"
  type="error"
  visible={true}
>
  <div>
    <div>
      test
    </div>
  </div>
</ToastContainer>
`;

exports[`Tabs should match the snapshot for default props 1`] = `
<ToastContainer
  align="middle"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  type="error"
  visible={true}
>
  <div>
    <div>
      test
    </div>
  </div>
</ToastContainer>
`;

exports[`Tabs should match the snapshot not visible 1`] = `
<ToastContainer
  align="middle"
  position="fixed"
  theme={
    Object {
      "error": Object {
        "main": Object {
          "RGB": "220,16,16",
        },
      },
    }
  }
  top="20px"
  type="error"
  visible={false}
>
  <div>
    <div>
      test
    </div>
  </div>
</ToastContainer>
`;
